From 177daad994dfa69aadbf899decf312e9c563b0d1 Mon Sep 17 00:00:00 2001
From: Takashi Kajinami <tkajinam@redhat.com>
Date: Wed, 16 Mar 2022 17:12:19 +0900
Subject: [PATCH] Fix compatibility with oslo.context >= 4.0.0

The tenant argument of RequestContext is no longer available since
oslo.context >= 4.0.0 . This change fixes the compatibility issue
caused by that removal.

NOTE:
Because scenario jobs are still broken with this change, these jobs
are made non-voting temporally to unblock the other jobs. Currently
this issue is blocking package build in RDO.

Story: 2009921
Task: 44779
Change-Id: I1bc81b3c13d2c08bc175d0d4f4365de7b4f71cf9
(cherry picked from commit d3ba5357c3bdfbfe542c047f1b347090aec29273)
---
 .zuul.yaml                                     | 12 ++++++++----
 sahara/context.py                              | 10 +++++-----
 sahara/tests/unit/plugins/test_provisioning.py |  4 ++--
 3 files changed, 15 insertions(+), 11 deletions(-)

diff --git a/.zuul.yaml b/.zuul.yaml
index e753d0e7..5764db54 100644
--- a/.zuul.yaml
+++ b/.zuul.yaml
@@ -10,8 +10,10 @@
       jobs:
         - openstack-tox-pylint:
             voting: false
-        - sahara-tests-scenario
-        - sahara-tests-scenario-v2
+        - sahara-tests-scenario:
+            voting: false
+        - sahara-tests-scenario-v2:
+            voting: false
         - sahara-tests-tempest
         - sahara-tests-tempest-v2
         - openstack-tox-cover:
@@ -23,8 +25,10 @@
     gate:
       queue: sahara
       jobs:
-        - sahara-tests-scenario
-        - sahara-tests-scenario-v2
+        - sahara-tests-scenario:
+            voting: false
+        - sahara-tests-scenario-v2:
+            voting: false
         - sahara-tests-tempest
         - sahara-tests-tempest-v2
         # - sahara-grenade
diff --git a/sahara/context.py b/sahara/context.py
index 07545d5e..2727144f 100644
--- a/sahara/context.py
+++ b/sahara/context.py
@@ -56,7 +56,7 @@ class Context(context.RequestContext):
 
         super(Context, self).__init__(auth_token=auth_token,
                                       user=user_id,
-                                      tenant=tenant_id,
+                                      project_id=tenant_id,
                                       is_admin=is_admin,
                                       resource_uuid=resource_uuid,
                                       request_id=request_id,
@@ -105,8 +105,8 @@ class Context(context.RequestContext):
         return d
 
     def is_auth_capable(self):
-        return (self.service_catalog and self.auth_token and self.tenant and
-                self.user_id)
+        return (self.service_catalog and self.auth_token and self.project_id
+                and self.user_id)
 
     # NOTE(adrienverge): The Context class uses the 'user' and 'tenant'
     # properties internally (inherited from oslo_context), but Sahara code
@@ -121,11 +121,11 @@ class Context(context.RequestContext):
 
     @property
     def tenant_id(self):
-        return self.tenant
+        return self.project_id
 
     @tenant_id.setter
     def tenant_id(self, value):
-        self.tenant = value
+        self.project_id = value
 
 
 def get_admin_context():
diff --git a/sahara/tests/unit/plugins/test_provisioning.py b/sahara/tests/unit/plugins/test_provisioning.py
index 17c73ec5..d0344723 100644
--- a/sahara/tests/unit/plugins/test_provisioning.py
+++ b/sahara/tests/unit/plugins/test_provisioning.py
@@ -110,13 +110,13 @@ class TestPluginDataCRUD(base.SaharaWithDbTestCase):
         self.assertIsNotNone(raised)
 
         # not duplicated entry, other tenant
-        ctx.tenant = "tenant_2"
+        ctx.project_id = "tenant_2"
         res = conductor.plugin_create(ctx, {'name': 'fake'})
         conductor.plugin_create(ctx, {'name': 'guy'})
         self.assertIsNotNone(res)
         self.assertEqual(2, len(conductor.plugin_get_all(ctx)))
 
-        ctx.tenant = "tenant_1"
+        ctx.project_id = "tenant_1"
 
         data = conductor.plugin_get(ctx, 'fake')
         self.assertEqual('fake', data['name'])
-- 
2.35.1

